src/GraphControl/Control/Grapher.cpp

```diff
@@ -192,7 +192,16 @@ namespace GraphControl
             return;
         }
 
-        PlotGraph(true);
+        bool keepCurrentView = true;
+
+        // If the equation has changed, the IsLineEnabled state is reset.
```

Comment:
I don't see IsLineEnabled getting reset when the equation is changed

Reply:
This is fixed. It should be working now. Let me know what you think of the changes.